### PR TITLE
`Separator` - Convert to TypeScript

### PR DESCRIPTION
**Pull request by Dhaulagiri**
### :pushpin: Summary

Converts the Separator comopnent to TS

### :link: External links

Jira ticket: [HDS-2710](https://hashicorp.atlassian.net/browse/HDS-2710)

***

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

[HDS-2710]: https://hashicorp.atlassian.net/browse/HDS-2710?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by didoo — APPROVED** [review]
Pre-approving, but don't forget to remove `{{! @glint-nocheck: not typesafe yet }}` from the template file

**Comment by Dhaulagiri** [comment]
🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦  i was even reading the instructions as I went through this.

File: .changeset/stupid-news-marry.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@hashicorp/design-system-components": minor
+---
+
+`Separator` - Convert component to TypeScript
```
Comment (alex-ju) [inline .changeset/stupid-news-marry.md:5]: We seem to be aiming for past tense

```suggestion
`Separator` - Converted component to TypeScript
```

Comment (Dhaulagiri) [reply .changeset/stupid-news-marry.md]: 👍  made this change on my initial commit

**Pull request merged**